### ISSUE TITLE
Update operations fields with omitempty when tx is failed

### ISSUE THREAD

**Issue by bartekn** [stellar/go#3869]
Some operation details data may not be be available if tx is failed. For example, LP meta will be empty for `LiquidityPoolDeposit`. In such case the operation details with information that is unavailable should be hidden.

**Comment by 2opremio** [comment]
Wouldn't it be more cohesive/consistent to show empty fields? We can document that the fields will be empty when the transactions fails

Although .... maybe it makes operations harder to parse.

**Comment by bartekn** [comment]
I checked the API again and I think you are right we can do omitempty for some and zero values for others. The only fields that require meta are: 
* for deposit:
    * LP assets - which probably should be hidden,
    * `reserves_deposited` which can be 0 because it's 0 for failed txs.
* for withdrawal:
    * LP assets - which probably should be also hidden,
    * `reserves_received` which can be 0 because it's 0 for failed txs.

**Comment by 2opremio** [comment]
I think that it would be ideal to set the amounts to 0 when the transaction fails. e.g.

```
{
  "id": "3697472920621057",
  "paging_token": "3697472920621057",
  "transaction_successful": false,
  "source_account": "GBB4JST32UWKOLGYYSCEYBHBCOFL2TGBHDVOMZP462ET4ZRD4ULA7S2L",
  "type": "liquidity_pool_deposit",
  "type_i": 22,
  "created_at": "2021-11-18T03:47:47Z",
  "transaction_hash": "43ed5ce19190822ec080b67c3ccbab36a56bc34102b1a21d3ee690ed3bc23378",
  "liquidity_pool_id": "abcdef",
  "reserves_max": [
    {
      "asset": "JPY:GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM",
      "amount": "1000.0000005"
    },
    {
      "asset": "EURT:GAP5LETOV6YIE62YAM56STDANPRDO7ZFDBGSNHJQIYGGKSMOZAHOOS2S",
      "amount": "3000.0000005"
    }
  ],
  "min_price": "0.2680000",
  "min_price_r": {
    "n": 67,
    "d": 250
  },
  "max_price": "0.3680000",
  "max_price_r": {
    "n": 73,
    "d": 250
  },
  "reserves_deposited": [
    {
      "asset": "JPY:GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM",
      "amount": "0"
    },
    {
      "asset": "EURT:GAP5LETOV6YIE62YAM56STDANPRDO7ZFDBGSNHJQIYGGKSMOZAHOOS2S",
      "amount": "0"
    }
  ],
  "shares_received": "0"
}
```

and

```
{
  "id": "3697472920621057",
  "paging_token": "3697472920621057",
  "transaction_successful": true,
  "source_account": "GBB4JST32UWKOLGYYSCEYBHBCOFL2TGBHDVOMZP462ET4ZRD4ULA7S2L",
  "type": "liquidity_pool_withdraw",
  "type_i": 23,
  "created_at": "2021-11-18T03:47:47Z",
  "transaction_hash": "43ed5ce19190822ec080b67c3ccbab36a56bc34102b1a21d3ee690ed3bc23378",
  "liquidity_pool_id": "abcdef",
  "reserves_min": [
    {
      "asset": "JPY:GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM",
      "min_amount": "1000.0000005"
    },
    {
      "asset": "EURT:GAP5LETOV6YIE62YAM56STDANPRDO7ZFDBGSNHJQIYGGKSMOZAHOOS2S",
      "min_amount": "3000.0000005"
    }
  ],
  "shares": "200",
  "reserves_received": [
    {
      "asset": "JPY:GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM",
      "amount": "0"
    },
    {
      "asset": "EURT:GAP5LETOV6YIE62YAM56STDANPRDO7ZFDBGSNHJQIYGGKSMOZAHOOS2S",
      "amount": "0"
    }
  ]
}
```

However, we would need to obtain the assets ... which may not even be available in our database if the transaction contains the liquidity pool creation (or if the liquidity_pool_id in the operation is wrong).

Thus, I think we should just add omitempty to every field which may not be obtainable when the transaction fails.

**Comment by 2opremio** [comment]
Another option is to leave the assets as an empty string (or omit them),  when they are unknown, that would be:

```
{
  "id": "3697472920621057",
  "paging_token": "3697472920621057",
  "transaction_successful": false,
  "source_account": "GBB4JST32UWKOLGYYSCEYBHBCOFL2TGBHDVOMZP462ET4ZRD4ULA7S2L",
  "type": "liquidity_pool_deposit",
  "type_i": 22,
  "created_at": "2021-11-18T03:47:47Z",
  "transaction_hash": "43ed5ce19190822ec080b67c3ccbab36a56bc34102b1a21d3ee690ed3bc23378",
  "liquidity_pool_id": "abcdef",
  "reserves_max": [
    {
      "asset": "",
      "amount": "1000.0000005"
    },
    {
      "asset": "",
      "amount": "3000.0000005"
    }
  ],
  "min_price": "0.2680000",
  "min_price_r": {
    "n": 67,
    "d": 250
  },
  "max_price": "0.3680000",
  "max_price_r": {
    "n": 73,
    "d": 250
  },
  "reserves_deposited": [
    {
      "asset": "",
      "amount": "0"
    },
    {
      "asset": "",
      "amount": "0"
    }
  ],
  "shares_received": "0"
}
```

```
{
  "id": "3697472920621057",
  "paging_token": "3697472920621057",
  "transaction_successful": true,
  "source_account": "GBB4JST32UWKOLGYYSCEYBHBCOFL2TGBHDVOMZP462ET4ZRD4ULA7S2L",
  "type": "liquidity_pool_withdraw",
  "type_i": 23,
  "created_at": "2021-11-18T03:47:47Z",
  "transaction_hash": "43ed5ce19190822ec080b67c3ccbab36a56bc34102b1a21d3ee690ed3bc23378",
  "liquidity_pool_id": "abcdef",
  "reserves_min": [
    {
      "asset": "",
      "amount": "1000.0000005"
    },
    {
      "asset": "",
      "amount": "3000.0000005"
    }
  ],
  "shares": "200",
  "reserves_received": [
    {
      "asset": "",
      "amount": "0"
    },
    {
      "asset": "",
      "amount": "0"
    }
  ]
}
```

This provides the highest amount of information, I think I would go for this.

**Comment by bartekn** [comment]
Fixed in #3880.
